Applications/spire/source/time_and_sales/time_and_sales_window.cpp

```diff
@@ -32,19 +42,211 @@ time_and_sales_window::time_and_sales_window(
+      padding-left: %1px;
+      padding-right: %1px;
+    }
+
+    QHeaderView::section::first {
+      background: none;
+      background-color: #FFFFFF;
+    })").arg(scale_width(8)));
+  m_table->setStyleSheet(QString(R"(
+    QTableView {
+      border: none;
+      gridline-color: #C8C8C8;
+    }
+
+    QScrollBar::horizontal {
+      height: %1px;
+    }
+
+    QScrollBar::vertical {
+      width: %2px;
+    }
+
+    QScrollBar::handle {
+      background-color: #C8C8C8;
+    }
+
+    QScrollBar::handle:horizontal {
+      min-width: %3px;
+    }
+
+    QScrollBar::handle:vertical {
+      min-height: %4px;
+    }
+
+    QScrollBar::add-line, QScrollBar::sub-line,
+    QScrollBar::add-page, QScrollBar::sub-page {
+      background: none;
+      border: none;
+      height: 0px;
+      width: 0px;
+    })").arg(scale_height(12)).arg(scale_height(12))
+        .arg(scale_width(30)).arg(scale_height(30)));
+  m_table->hide();
+  layout->addWidget(m_table);
+  m_volume_label = new QLabel(tr("Volume: 0"), this);
+  m_volume_label->setFocusPolicy(Qt::NoFocus);
+  m_volume_label->setFixedHeight(scale_height(20));
+  m_volume_label->setStyleSheet(QString(R"(
+    background-color: #F5F5F5;
+    color: #333333;
+    font-family: Roboto;
+    font-size: %1px;
+    font-weight: 550;
+    padding-left: %2px;)").arg(scale_height(10)).arg(scale_width(8)));
+  layout->addWidget(m_volume_label);
+  m_context_menu = new QMenu(this);
+  auto properties_action = new QAction(tr("Properties"), m_context_menu);
+  connect(properties_action, &QAction::triggered,
+    [=] { show_properties_dialog(); });
+  m_context_menu->addAction(properties_action);
+  auto book_view_action = new QAction(tr("Link Book View"), m_context_menu);
+  m_context_menu->addAction(book_view_action);
+  auto chart_action = new QAction(tr("Link Chart"), m_context_menu);
+  m_context_menu->addAction(chart_action);
+  auto export_action = new QAction(tr("Export Table"), m_context_menu);
+  m_context_menu->addAction(export_action);
+  m_context_menu->setFixedWidth(scale_width(140));
+  m_context_menu->setWindowFlag(Qt::NoDropShadowWindowHint);
+  m_context_menu_shadow = std::make_unique<drop_shadow>(
+    true, true, m_context_menu);
+  m_context_menu->setStyleSheet(QString(R"(
+    QMenu {
+      background-color: #FFFFFF;
+      border: %1px solid #A0A0A0 %2px solid #A0A0A0;
+      color: #000000;
+      font-family: Roboto;
+      font-size: %3px;
+      padding: %4px 0px;
+    }
+
+    QMenu::item {
+      padding: %5px 0px %5px %6px;
+    }
+
+    QMenu::item:selected, QMenu::item:hover {
+      background-color: #8D78EC;
+      color: #FFFFFF;
+    })")
+    .arg(scale_height(1)).arg(scale_width(1))
+    .arg(scale_height(12)).arg(scale_height(5))
+    .arg(scale_height(3)).arg(scale_width(8)));
+  set_model(std::make_shared<empty_time_and_sales_model>(Security()));
+  set_properties(properties);
+  m_properties_dialog = new time_and_sales_properties_dialog(properties, this);
```

Comment:
Properties dialog needs to be modal.

Reply:
Don't preconstruct this dialog and show/hide it. Instead only create it when you actually need it.